### ISSUE TITLE
Dotnet trimmer testing

### ISSUE THREAD

**Issue by tranb3r** [TobiasBuchholz/Plugin.Firebase#144]
Hi,

When migrating playground/tests to dotnet maui, the `<AndroidLinkMode>Full</AndroidLinkMode>` property has been removed from the release build.
I don't know if it's been done on purpose, but I think you could add now the corresponding dotnet trimmer property: `<TrimMode>full</TrimMode>`.

Also, are you sure the `Preserve` attributes in the plugin code are still needed?
Maybe it's worth testing it again with dotnet trimmer. What do you think?

Thanks!

**Comment by tranb3r** [comment]
When trimming my application, android's `MyFirebaseMessagingService` is removed.
I can workaround that issue by disabling trimming for `Plugin.Firebase.CloudMessaging`.
But maybe this could be fixed in the plugin's code, by ensuring the service is always preserved?

**Comment by TobiasBuchholz** [comment]
Okay yes, I will add the `Preserve` attribute to the `MyFirebaseMessagingService` and release it with the next `Plugin.Firebase.CloudMessaging` version 👍

**Comment by tranb3r** [comment]
Ok but I'm not 100% sure that the `Preserve` attribute is still the way to preserve code in dotnet maui.
Can you confirm that?

**Comment by TobiasBuchholz** [comment]
No, I can't confirm that, I was just assuming it would work. What would be the alternative?

**Comment by Quaybe** [comment]
Disabling trimming completely until this is resolved. The workarounds no longer work for me.

**Comment by jonathanpeppers** [comment]
Until this one ships:
* https://github.com/dotnet/android/pull/9099

I found putting this in your app seems to workaround the issue:
```csharp
[DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(MyFirebaseMessagingService))]
```

Can anyone confirm if it solve the problem for them?

Is this only happening when using `TrimMode=full`, or does it also occur with default settings?

**Comment by dartasen** [comment]
@jonathanpeppers Seems like it's working just fine with DynamicDependency attribute !
Even if Preserve attribute is flagged as Obsolete, still being usable is a bit confusing since it seems to have stopped working properly

**Comment by jonathanpeppers** [comment]
No, I don't think we can remove `[ObsoleteAttribute]` as that would be a breaking change. Random Xamarin.Android class libraries could throw `TypeLoadException` (or possibly even more inscrutable runtime errors).

It does seem like we could add `[Obsolete(error: true)]` in .NET 9, so you'd get a build error in new source code if you used it. I'll think on it.

**Comment by LennoxP90** [comment]
there still is an issue with trimming and requires 
[DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(MyFirebaseMessagingService))]

even with PR9099 shipped

**Comment by elfiservice** [comment]
I still have problems to build Android Release version
my class is being Trimmed 

i'm using .net 9  sdk 9.0.304
Plugin.Firebase.CloudMessaging nuget  v 3.1.2

if disabling trimming it works.

```CSharp
[Service(Exported = false)]
[IntentFilter(["com.google.firebase.MESSAGING_EVENT"])]
public class PushNotificationFirebaseMessagingService : FirebaseMessagingService

```

@tranb3r  Were you able to fix that problem? ?

**Comment by tranb3r** [comment]
> [@tranb3r](https://github.com/tranb3r) Were you able to fix that problem? ?

Sorry, I'm not using this library anymore.

**Comment by Mike-E-angelo** [comment]
> I found a workaround reference from https://github.com/TobiasBuchholz/Plugin.Firebase/issues/144#issuecomment-3825432656

Wow @elfiservice thank you so much for posting this.  I have been battling this issue in one form or another for the past week and was about to give up and turn off trimming altogether.  I, too, was running into this issue and there is no way I would have found out about it.  I got that remove in there and it saved the day.  THANK YOU! 🙏

**Comment by AdamEssenmacher** [comment]
I took another look at this while checking the current Cloud Messaging implementation.

The original concern about `Plugin.Firebase.CloudMessaging.Platforms.Android.MyFirebaseMessagingService` being trimmed under `TrimMode=full` appears to be covered by the upstream .NET Android fix in dotnet/android#9099, which preserves Android Java-object types with JNI-name-provider attributes. With the current .NET 9 Android SDK, a Release `TrimMode=full` build keeps the plugin service in the generated callable-wrapper map and Android manifest.

I opened #614 as a draft PR to add regression coverage for this repo. It builds the Android integration-test host with `TrimMode=full` and verifies that the plugin's `MyFirebaseMessagingService` is still preserved, registered in the manifest, marked `android:exported="false"`, and wired to `com.google.firebase.MESSAGING_EVENT`. The PR also documents the default service behavior and the custom-service/removal path for apps that need to provide their own `FirebaseMessagingService`.
